client/flutter/lib/utils/debounce_handler.dart

```diff
@@ -0,0 +1,19 @@
+
+import 'dart:async';
+import 'package:flutter/material.dart';
+
+/// Handle debounce when we click multiple times quickly.
+class DebounceHandler {
```

Comment:
Is this the best practice generally, or your own solution?

Reply:
Hi guys! Why don't we use streams and the debounce extension of rxdart?

Reply:
I am a huge fan of RxDart and as we add complexity to the app I would like to see us introduce it where appropriate.  However if we need button debouncing in these simple cases it might just be best to do something ad-hoc for now.  Rather than a timer I would probably just add a `fooEnabled` state variable to the containing class to govern the button state.  This is about as simple as it gets and can be controlled by the logic as necessary.  If we get into more complex situations composing data sources and publishing state to the entire app we should definitely use streams and probably RxDart.